src/MxcImageProvider.cpp

```diff
@@ -108,6 +111,12 @@ MxcImageProvider::download(const QString &id,
     if (temp != infos.end())
         encryptionInfo = *temp;
 
+    if (cachedThumbnails.contains(id))
```

Comment:
The cache key should probably include the size.

Reply:
Oh yeah, didn't think about that